cmd/spruce/main.go

```diff
@@ -235,7 +235,8 @@ func parseYAML(data []byte) (map[interface{}]interface{}, error) {
 		if _, arrayErr := y.Array(); arrayErr == nil {
 			return nil, RootIsArrayError{msg: ansi.Sprintf("@R{Root of YAML document is not a hash/map}: %s\n", err)}
 		}
-		return nil, ansi.Errorf("@R{Root of YAML document is not a hash/map}: %s\n", err.Error())
+		ansi.Errorf("@R{Root of YAML document is not a hash/map, using empty map instead.}\n")
+		doc := make(map[interface{}]interface{})
```

Comment:
Can this be done in a way where it only makes an empty doc if the root document is completely empty? 

Also, the call to ansi.Errorf() here does not dow hat you're expecting. it generates and returns a golang error struct, but since it isn't saved to a variable, the error goes nowhere. The `:=` on 239 also re-initializes `doc` in a lower scope, and the code fails to compile. You probably want `=` instead.

I'm going to add some tests on the master branch to make testing this change easier if you want to pull those changes in

Reply:
1. I would agree that handling only empty docs in this way would be preferred, but I have trouble defining what an empty doc is. E.g. _bosh-cli runtime-config_ seems to return a newline when bosh director has no runtime config. If so, we could try to handle only completely empty docs and docs with newline this way (though I am not sure other tools would work similar to bosh-cli when returning no meaningful content). Or we can try raising an issue with bosh-cli to claim they should return nothing in stdout if there is no meaningful content to return.
2. I am sorry, I have to admit my main intention was to start the discussion. Of course, once we agree on the best approach I would be happy to use the tests you are suggesting to put in place and to nature this PR.

Reply:
No worries, I appreciate the conversation!

For empty yaml files, I would define that as anything a yaml parser returns as a nil object. So probably a file with nothing, a file with only white space, or a file with only yaml’s null keywords in it?

Reply:
Thanks, that sounds like a wonderful technical empty doc criterion. Let me go over the code with this definition in mind to see what would be the best way to implement it.